### PR TITLE
New NOT_WORKING machine

### PR DESCRIPTION
**Pull request by clawgrip**
--------------------------
Olimpic Hockey (EM Bubble Hockey) [Juan Romero, ClawGrip]

### CODE REVIEW

**Comment by clawgrip** [comment]
This set was dumped some years ago, and the dumper do not keep any PCB pics. It's for sure from an Inor Olimpic Hockey, and since Inor only used two kind of PCBs on these machines, the one at chexx.cpp and this one, it must be this one (see attached pic and https://media.recreativas.org/manuales/futbol-champ-manual.pdf, same PCB for another EM bubble game). 

The code seems to be Z80, so it should be OK, but I don't feel comfortable with the doubt. Is it possible for someone to check if the ROM reasonably matches the PCB hardware? @ajrhacker ?
The machine is rare enough to have it preserved on MAME.

Thanks a lot. Best regards: ClawGrip

pcb2_inor_hockey

**Comment by ajrhacker** [comment]
I haven't obtained the ROM yet. However, recreativas.org has a manual and PCB pictures for Olimpic Hockey that show that it runs on a completely platform using a 6502 CPU, 6522 VIA and National Semiconductor DigiTalker. Furthermore, it appears to be a license-built copy of the original Chexx (1983).

**Comment by clawgrip** [comment]
Yes, that's the older hardware, a direct clone of the Chexx PCB (see atached pic, I have two of them for dumping).
Inor later replaced the electronics with this newer 8080-based hardware, that isn't related to any ICE hardware, AFAIK.

I sent the ROM to The Dumping Union.

image

File: scripts/target/mame/arcade.lua [hunk 1]
```diff
@@ -4963,6 +4963,7 @@ files {
 	MAME_DIR .. "src/mame/drivers/ichiban.cpp",
 	MAME_DIR .. "src/mame/drivers/imolagp.cpp",
 	MAME_DIR .. "src/mame/drivers/intrscti.cpp",
+	MAME_DIR .. "src/mame/drivers/inorbubble.cpp",
 	MAME_DIR .. "src/mame/drivers/island.cpp",
```
Comment (cuavas) [inline scripts/target/mame/arcade.lua:4967]: This has got out of alphabetical order.

File: src/mame/drivers/inorbubble.cpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	required_device<cpu_device> m_maincpu;
+};
+
+INPUT_PORTS_START(inorbubble)
+INPUT_PORTS_END
+
+void inorbubble_state::inorbubble(machine_config &config)
+{
+	// Basic machine hardware
+	I8080(config, m_maincpu, 4'000'000); // Unknown clock // AMD 8080 CPU, exact model unknown
```
Comment (cuavas) [inline src/mame/drivers/inorbubble.cpp:46]: You said “the code seems to be Z80” in the description – if it’s using Z80 features, it won’t work on an 8080.

Comment (clawgrip) [reply src/mame/drivers/inorbubble.cpp:46]: That's why I'm asking for help, this is far beyond my skills.

File: scripts/target/mame/arcade.lua (hunk 1 above)
Comment (clawgrip) [reply scripts/target/mame/arcade.lua]: Uuups! Fixed, thanks!

**Comment by ajrhacker** [comment]
I've finally been able to take a look. It definitely seems like Z80 code, using instructions not available on 8080 or 8085. That said, the dump is not remotely good. It has a large number of single-bit glitches, which are especially obvious in the part starting at 0x2730 that should probably be just 0xFF fill.

**Comment by clawgrip** [comment]
Thanks a lot!!!

I’ll try to find another machine and dump it by myself.

**Comment by cuavas** [comment]
>It definitely seems like Z80 code, using instructions not available on 8080 or 8085.

This strongly suggests it ran on a different board.  The Z80 uses a different bus interface to the 8080.  It pays to actually document things.

**Comment by happppp** [comment]
different board from what? The Futbol manual says Z80 and pcb photo in it matches this thing.

**Comment by clawgrip** [comment]
> different board from what? The Futbol manual says Z80 and pcb photo in it matches this thing.

I thought it to be an 8080 just because of the color picture I found.
It’s not easy to work with a dump made some years ago from someone else, without any further info.
Sorry about that.

Anyway, being a bad dump… I’ll try to locate another machine. Not easy, but who knows.

**Comment by ajrhacker** [comment]
8080-based systems generally use the Intel 8224 two-phase clock generator (or one of its clones); it's either that or a complicated assortment of TTL including level converters. It's not a totally reliable indicator; there are a few Z80-based boards that are know to use a 8224, perhaps for a TMS5501 or S-100 bus (both of which were originally designed around the 8080). On the software side, the Z80 was designed to be backward-compatible with 8080 machine code, and a few Z80 programs seem to have been developed using only 8080 opcodes.

It might be possible that a Mostek MK3880 (a common second source of the Z80 CPU) was mistaken for an 8080. Anyway, here's a different photo and schematic of "Olimpic Hockey" that unmistakably shows that the original Inor hardware, like most Spanish hardware of the 1980s, is Z80-based:

https://www.flipperfrance.com/threads/super-chexx-bubble-hockey.13287/

**Comment by clawgrip** [comment]
Thanks a lot @ajrhacker !

Yes, two sources points to a Z80, while it was just me thinking that it is an AMD 9080 because it looks like it on the low res picture.

So, the ROM really IS from that machine, but it's a bad dump, isn't it?
Do you think it's worth adding it on a skeleton being bad? I really don't, that's why I've closed the PR...

Thanks again!

**Pull request closed without merging**